.github/workflows/main.yml

```diff
@@ -0,0 +1,394 @@
+        if: steps.cache-apt.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get update
+          sudo apt-get install --download-only -y --no-install-recommends \
+            asciidoctor \
+            cmake \
+            dbus-x11 \
+            gettext \
+            gir1.2-gtk-3.0 \
+            gir1.2-pango-1.0 \
+            git \
+            libdbus-1-dev \
+            libgirepository1.0-dev \
+            libnotify-bin \
+            libpango1.0-dev \
+            libstartup-notification0-dev \
+            libx11-xcb-dev \
+            libxcb-cursor-dev \
+            libxcb-icccm4-dev \
+            libxcb-keysyms1-dev \
+            libxcb-randr0-dev \
+            libxcb-shape0-dev \
+            libxcb-util0-dev \
+            libxcb-xfixes0-dev \
+            libxcb-xinerama0-dev \
+            libxcb-xkb-dev \
+            libxcb-xrm-dev \
+            libxcb-xtest0-dev \
+            libxdg-basedir-dev \
+            libxkbcommon-dev \
+            libxkbcommon-x11-dev \
+            wmctrl \
+            x11-apps \
+            xcb-proto \
+            xdotool \
+            xorg \
+            xserver-xephyr \
+            xterm \
+            xutils-dev \
+            xvfb \
+            zsh \
+            lua-discount
+
+      - name: Install downloaded packages
+        run: sudo dpkg -i /var/cache/apt/archives/*.deb
+
+      - name: Install Lua packages
+        run: |
+          if [ "${{ matrix.lua_name }}" = "luajit" ]; then
+            sudo apt-get install libluajit-5.1-dev luajit
+          fi
+          sudo apt-get install liblua${{ matrix.lua_version }}-dev lua${{ matrix.lua_version }}
+
+      - name: Cache luarocks
+        id: cache-luarocks
+        uses: actions/cache@v2
+        with:
+          path: /tmp/luarocks
+          # The build input for luarocks changes per test, so we need separate caches
+          key: ${{ github.workflow }}-${{ runner.os }}-${{ matrix.test_name }}-luarocks-3.5.0
+
+      - name: Install fresh Luarocks
+        if: steps.cache-luarocks.outputs.cache-hit != 'true'
+        run: |
+          wget -O /tmp/luarocks.tar.gz https://github.com/luarocks/luarocks/archive/v3.5.0.tar.gz
+          mkdir /tmp/luarocks
+          tar -xf /tmp/luarocks.tar.gz -C /tmp/luarocks --strip-components=1
+          cd /tmp/luarocks
+          ./configure --lua-version=${{ matrix.lua_version }} --with-lua-include=${LUAINCLUDE} ${{ matrix.luarocks_args }}
+          make build
+          sudo make install
+
+      - name: Install cached Luarocks
+        if: steps.cache-luarocks.outputs.cache-hit == 'true'
+        run: |
+          cd /tmp/luarocks
+          sudo make install
+
+      - name: Cache xcb-errors
+        id: cache-xcb-errors
+        uses: actions/cache@v2
+        with:
+          path: /tmp/xcb-errors
+          # xcb-errors doesn't have versioned releases, so we checkout and cache by commit.
+          # Make sure to keep this in sync with the `git checkout` below.
```

Comment:
What exactly does this mean and what is wrong with https://xcb.freedesktop.org/dist/xcb-util-errors-1.0.tar.bz2? (Found by looking at what Arch does: https://github.com/archlinux/svntogit-community/blob/55a631b407cc54dd60a85328c8fc1f60468c5dbf/trunk/PKGBUILD#L13 )

Reply:
The Travis job simply cloned `master`. To pin the version, I checked https://gitlab.freedesktop.org/xorg/lib/libxcb-errors/-/releases, but found it to be empty. I did forget to check the tags, apparently.
So I used the one revision that Git always offers.